### PR TITLE
Playlist functionality added with API endpoints, controller functionality, and db modeling

### PR DESCRIPTION
**Pull request by PrinceOm**

### CODE REVIEW

**Comment by underaj** [comment]
LGTM

File: api/playlist/index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const express = require('express');
+const controller = require('./playlist.controller');
+
+const router = express.Router();
+
+router.get('/', controller.lists);
+router.get('/:playlistId', controller.show);
+router.post('/', controller.create);
```
Comment (dting) [inline api/playlist/index.js]: lets put this above the line above.

File: api/playlist/playlist.controller.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+controller.lists = (req, res) => {
+  Playlist.find({}).exec()
+    .then(playlists => res.send(playlists))
+    .catch(handleError(res));
+};
+
+controller.show = (req, res) => {
+  const id = req.params.playlistId;
+  Playlist.findOne({ id }).exec()
```
Comment (dting) [inline api/playlist/playlist.controller.js]: We can use:
```
Playlist.findOneById(id).exec()
```

File: api/playlist/playlist.controller.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    .catch(handleError(res));
+};
+
+controller.removePodcast = (req, res) => {
+  const { playlistId, podcastId } = req.params;
+  Playlist.update(
+    { _id: playlistId },
+    { $pull: { podcasts: podcastId } }
+    ).exec()
+    .then(res.send('successfully removed'))
```
Comment (dting) [inline api/playlist/playlist.controller.js]: I think we want to send a different status code. Look up what is usually sent when deleting

File: api/playlist/index.js (hunk 1 above)
Comment (PrinceOm) [inline api/playlist/index.js]: Done

File: api/playlist/playlist.controller.js (hunk 2 above)
Comment (PrinceOm) [inline api/playlist/playlist.controller.js]: Done

File: api/playlist/playlist.controller.js (hunk 3 above)
Comment (PrinceOm) [inline api/playlist/playlist.controller.js]: Done

File: api/playlist/playlist.controller.js [hunk 4]
```diff
@@ -3,45 +3,51 @@ const Playlist = require('./playlist.model');
...
-	Playlist.findOne({id}).exec()
-		.then(list => Playlist.podcasts.push(p_id))
-		.then(response => res.send(response))
-		.catch(handleError(res));
+controller.addPodcast = (req, res) => {
+  const { playlistId, podcastId } = req.params;
+  Playlist.findById(playlistId).exec()
+    .then(list => {
+      list.podcasts.push(podcastId);
+      list.save();
```
Comment (dting) [inline api/playlist/playlist.controller.js]: This needs to be returned for the next `.then` to wait for the save to complete.

File: api/playlist/playlist.controller.js [hunk 5]
```diff
@@ -3,45 +3,51 @@ const Playlist = require('./playlist.model');
...
-	Playlist.findOne({id}).exec()
-		.then(list => console.log("need to delete playlist", list))
-		.then(response => res.send(response))
+controller.removePodcast = (req, res) => {
+  const { playlistId, podcastId } = req.params;
+  Playlist.update(
+    { _id: playlistId },
+    { $pull: { podcasts: podcastId } }
+		).exec()
+		.then(response => response.nModified > 0 ? res.sendStatus(202) : res.send('podcast not found'))
```
Comment (dting) [inline api/playlist/playlist.controller.js]: Does this pass the linter? I don't think it likes these ternaries.

Comment (dting) [inline api/playlist/playlist.controller.js]: There are tabs in this code.

Comment (PrinceOm) [inline api/playlist/playlist.controller.js]: Done.

File: api/playlist/playlist.controller.js (hunk 4 above)
Comment (PrinceOm) [inline api/playlist/playlist.controller.js]: Done

**Comment by dting** [comment]
LGTM.

**Pull request merged**